tensorflow/contrib/verbs/rdma.cc

```diff
@@ -471,8 +477,12 @@ void RdmaChannel::Connect(const RdmaAddress& remoteAddr) {
     attr.dest_qp_num = remoteAddr.qpn;
     attr.rq_psn = remoteAddr.psn;
     attr.max_dest_rd_atomic = 1;
-    attr.min_rnr_timer = 12;
-    attr.ah_attr.is_global = 0;
+    attr.min_rnr_timer = 12;   
+    attr.ah_attr.is_global = 1;
+    attr.ah_attr.grh.dgid.global.subnet_prefix = remoteAddr.snp;
+    attr.ah_attr.grh.dgid.global.interface_id = remoteAddr.iid;
+    attr.ah_attr.grh.flow_label = 0;
+    attr.ah_attr.grh.hop_limit = 3; 
     attr.ah_attr.dlid = remoteAddr.lid;
```

Comment:
@junshi15 - you can use ah_attr.is_global=1 in IB, it's not a must since since you have LID (Assuming L2 IB fabric)
In RoCE it's a must since the MAC is calculated from the GID. 

@il-ch - Thanks for the patch. Just curious about the hop_limit=3.
Usually it 1. Do you have L3 (router) Infiniband fabric?